pkg/service/collector.go

```diff
@@ -104,6 +110,31 @@ func GetNameForHeadlessCollectorService(jaeger *v1.Jaeger) string {
 	return util.DNSName(util.Truncate("%s-collector-headless", 63, jaeger.Name))
 }
 
+// GetPortForCollectorService returns the query service port number for this Jaeger instance
+func GetPortForCollectorService(jaeger *v1.Jaeger) int {
+	if jaeger.Spec.Collector.Ingress.Security == v1.IngressSecurityOAuthProxy {
+		return 443
+	}
+	//TODO: add a different port number
+	return 16686
```

Comment:
You will need to expose the different supported formats

Reply:
I am not sure what you mean by it. Can you point me to different exposed formats or any PR that has done it?

Reply:
The exposed [ports are listed in the documentation](https://www.jaegertracing.io/docs/1.47/getting-started/). If you expose the collector through Ingress, it would be needed to expose the different services for the protocols accepted by the collector

Reply:
From what I understand looking at the code setting ports in [service ports](https://github.com/jaegertracing/jaeger-operator/blob/main/pkg/service/collector.go#L46) should expose the ports like we are doing in [pkg/service/query.go](https://github.com/jaegertracing/jaeger-operator/blob/main/pkg/service/query.go#L28). We are already exposing the admin and grpc ports in collector. For service, we expose the port provided else we need some default port like we use [16686](https://github.com/jaegertracing/jaeger-operator/blob/main/pkg/service/query.go#L94) for query service.

Reply:
oh ok! Got it!